src/cascadia/TerminalSettingsModel/TerminalSettings.cpp

```diff
@@ -88,53 +96,100 @@ namespace winrt::Microsoft::Terminal::Settings::Model::implementation
     //     StartingDirectory) in this object to override the settings directly from
     //     the profile.
     // - keybindings: the keybinding handler
-    Model::TerminalSettings TerminalSettings::CreateWithNewTerminalArgs(const Model::CascadiaSettings& appSettings,
-                                                                        const Model::NewTerminalArgs& newTerminalArgs,
-                                                                        const IKeyBindings& keybindings)
+    // Return Value:
+    // - the GUID of the created profile, and a fully initialized TerminalSettings object
+    Model::TerminalSettingsStruct TerminalSettings::CreateWithNewTerminalArgs(const CascadiaSettings& appSettings,
+                                                                              const NewTerminalArgs& newTerminalArgs,
+                                                                              const IKeyBindings& keybindings)
     {
         const guid profileGuid = appSettings.GetProfileForArgs(newTerminalArgs);
-        auto settings{ CreateWithProfileByID(appSettings, profileGuid, keybindings) };
+        auto settings = CreateWithProfileByID(appSettings, profileGuid, keybindings);

```

Comment:
since you're using the `defaultSettings` a lot, cache it in a variable instead of looking it up repeatedly.